pkg/azurePipelines/azurePipelines.go

```diff
@@ -125,7 +125,7 @@ func (p *AzurePipelines) CreatePipelineFiles(deployType string, draftConfig *con
 		return fmt.Errorf("error applying default variables: %w", err)
 	}
 
-	if err := osutil.CopyDir(p.pipelineTemplates, srcDir, p.dest, draftConfig, templateWriter); err != nil {
+	if err := osutil.CopyDirWithTemplates(p.pipelineTemplates, srcDir, p.dest, draftConfig, templateWriter); err != nil {
```

Comment:
we should have some kind of unit test proving that these templates are properly filled out with the right Draft variables. Do we have that?

I like the fixture pattern

Reply:
I have tested them manually and yes, all the variables are replaced correctly!

Reply:
we need a unit test to prove and ensure that

Reply:
I have updated the unit tests with the fixtures!